### PR TITLE
Update demo to run on M1 systems

### PR DESCRIPTION
**Pull request by esune**
Updated vue app Dockerfile to run on M1 Mac architecture, added retry handling fro ngrok tunnel detection

### CODE REVIEW

**Comment by esune** [comment]
I think I got it sorted out now. @popkinj mind testing on your machine if the `null` controller url is already an issue?
@WadeBarnes let me know if you see anything odd that I might have overlooked.

**Comment by swcurran** [comment]
Does this mean we now have a working M1?

File: docker/manage [hunk 1]
```diff
@@ -346,27 +352,50 @@ function setNgrokEndpoints() {
...
+    else
+      return 0
+    fi
+  }
+
+  local startTime=${SECONDS}
+  while ! extractUrl; do
+    printf "."
+    local duration=$(($SECONDS - $startTime))
+    if (( ${duration} >= 60 )); then
```
Comment (WadeBarnes) [inline docker/manage:385]: I would make the timeout configurable with a 60 second default.

**Review by WadeBarnes — CHANGES_REQUESTED** [review]

File: docker/manage (hunk 1 above)
Comment (esune) [reply docker/manage:385]: I thought about that, it seemed though like 60s would be the worst case scenario and in most cases it should really work after a 5-6 seconds so adding another configuration setting might be overkill?

**Comment by esune** [comment]
> Does this mean we now have a working M1?

Yep, it should work on M1 without issues now - unless I missed something of course 🤣

**Review by WadeBarnes — APPROVED** [review]

**Review by popkinj — APPROVED** [review]
This looks like it'll do the trick.

**Pull request merged**